### PR TITLE
Update colander to current best practices

### PR DESCRIPTION
**Pull request by bertjwregeer**
- Move source in `src`
- Move tests into `tests`
- Move from `setup.py` to `setup.cfg`
- Fixes up `MANIFEST.in`
- Introduces Github Actions
- Updates `.gitignore`
- Updates `tox.ini`
- Drops Python 3.4 support
- Drops Travis for doing builds

Closes https://github.com/Pylons/colander/issues/210

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,34 +1,70 @@
...
+classifiers =
+    Development Status :: 6 - Mature
+    Intended Audience :: Developers
+    Programming Language :: Python
+    Programming Language :: Python :: 2
+    Programming Language :: Python :: 2.7
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.5
+    Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
```
Comment (stevepiercy) [inline setup.cfg:19]:
```suggestion
    Programming Language :: Python :: 3.7
    Programming Language :: Python :: 3.8
```

Comment (stevepiercy) [inline setup.cfg:19]:
```suggestion
    Programming Language :: Python :: 3.7
    Programming Language :: Python :: 3.8
```

File: setup.cfg [hunk 2]
```diff
@@ -1,34 +1,70 @@
...
+    Programming Language :: Python :: Implementation :: CPython
+    Programming Language :: Python :: Implementation :: PyPy
+    Operating System :: OS Independent
```
Comment (stevepiercy) [inline setup.cfg:22]:
```suggestion
    Operating System :: OS Independent
    License :: Repoze Public License
```

File: setup.cfg [hunk 3]
```diff
@@ -1,34 +1,70 @@
...
+url = https://github.com/Pylons/colander
```
Comment (stevepiercy) [inline setup.cfg:23]:
```suggestion
url = https://github.com/Pylons/colander
project_urls =
    Documentation = https://docs.pylonsproject.org/projects/colander/en/latest/index.html
    Changelog = https://docs.pylonsproject.org/projects/colander/en/latest/changes.html
    Issue Tracker = https://github.com/Pylons/colander/issues
```

File: setup.cfg [hunk 4]
```diff
@@ -1,34 +1,70 @@
...
+author = Agendaless Consulting
+author_email = pylons-discuss@googlegroups.com
+maintainer = Pylons Project
+maintainer_email = pylons-discuss@googlegroups.com
 
-[easy_install]
-zip_ok = false
+[options]
+package_dir=
+    =src
```
Comment (stevepiercy) [inline setup.cfg:31]: Is that `=` a typo?

Comment (bertjwregeer) [reply setup.cfg:37]: No.

**Pull request merged**